api/controllers/CampaignBotController.js

```diff
@@ -207,32 +233,25 @@ class CampaignBotController {
   }
 
   /**
-   * Gets User from DS API if exists for given id, else creates new User.
-   * @param {string} id - DS User ID
+   * Gets User from DS API if exists for given type/id, else creates new User.
+   * @param {object} req
    * @return {object} - User model
    */
-  getUser(id) {
+  getUser(type, id) {
+    logger.debug(`getUser type:${type} id:${id}`);
+
     return app.locals.clients.northstar.Users
-      .get('id', id)
-      .then(user => {
-        if (!user) {
-          // TODO: Edge case where User ID saved to Mobile Commons profile
-          // is not found in DS API GET request.
-        }
-        const data = {
-          _id: id,
-          mobile: user.mobile,
-          first_name: user.firstName,
-          email: user.email,
-          phoenix_id: user.drupalID,
-          role: user.role,
-          campaigns: {},
-        };
-        return app.locals.db.users
-          .findOneAndUpdate({ _id: data._id }, data, {
-            upsert: true,
-            new: true,
-          });
+      .get(type, id)
+      .then((user) => {
+        logger.debug('northstar.Users.get success');
+
+        return this.cacheUser(user);
+      })
+      .catch((err) => {
+        logger.error(err);
```

Comment:
Logging this error (which we get if Northstar can't find the given User) breaks the app, so I'm going to remove it for merge.

The addition of https://github.com/DoSomething/northstar/pull/452 leads me to wonder if Northstar JS should return a null `user` if we get a 404 returned (previously we'd just get HTML)